pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -131,6 +131,8 @@ func (r *ReconcileOneAgent) Reconcile(request reconcile.Request) (reconcile.Resu
 	}
 	r.scheme.Default(instance)
 
+	instance.Labels = mergeLabels(instance.GetLabels(), buildLabels(instance.Name))
```

Comment:
```suggestion
	instance.Spec.Labels = mergeLabels(instance.Spec.Labels, buildLabels(instance.Name))
```

`instance.Labels` and `instance.GetLabels()` both return `.metadata.labels` and not `.spec.labels` AFAIK.

The same applies to `newDaemonSetForCR()` but that aside it should do.

Although perhaps we need to handle `instance.Spec.Labels == nil`